intermediate/challenge-1/app/api/api_v1/endpoints/users.py

```diff
@@ -0,0 +1,114 @@
+from typing import Any, List
+from fastapi import APIRouter, Body, Depends, HTTPException, BackgroundTasks
+from fastapi.encoders import jsonable_encoder
+from pydantic.networks import EmailStr
+from sqlalchemy.orm import Session
+import crud
+import models
+import schemas
+from api import deps
+from core.config import settings
+from utils import send_new_account_email
+
+router = APIRouter()
+
+
+@router.get("/", response_model=List[schemas.User])
+def read_users(
+        db: Session = Depends(deps.get_db),
+        current_user: models.User = Depends(deps.get_current_active_manager),
+) -> Any:
+    """
+    Retrieve users.
+    """
+    users = crud.user.get_all(db)
+    return users
+
+
+@router.post("/", response_model=schemas.User)
+def create_user(
+        *,
+        db: Session = Depends(deps.get_db),
+        user_in: schemas.UserCreate,
+        current_user: models.User = Depends(deps.get_current_active_manager),
+        background_tasks: BackgroundTasks
+) -> Any:
+    """
+    Create new user.
+    """
+    user = crud.user.get_by_email(db, email=user_in.email)
+    if user:
+        raise HTTPException(
+            status_code=400,
+            detail="The user with this username already exists in the system.",
+        )
+    user = crud.user.create(db, obj_in=user_in)
+    if settings.EMAILS_ENABLED and user_in.email:
+        background_tasks.add_task(send_new_account_email,
+                                  email_to=user_in.email, username=user_in.email, password=user_in.password
+                                  )
+    return user
+
+
+@router.put("/me", response_model=schemas.User)
+def update_user_me(
+        *,
+        db: Session = Depends(deps.get_db),
+        password: str = Body(None),
+        full_name: str = Body(None),

```

Comment:
Is there any reason for not using the UserUpdate object as input parameter instead, similarly to how it is used when updated a User by ID ?

Reply:
UserUpdate object is used by PUT "/{user_id}" when a manager wants to change another user's information (e.g. upgrade a user's role from employee to manager), which does is not the case of the PUT "/me" which is simply invoked when the user wants to update his own information (name, password, email)